src/Type/TypehintHelper.php

```diff
@@ -201,8 +201,11 @@ public static function decideType(
 			}
 
 			if (
-				(!$phpDocType instanceof NeverType || ($type instanceof MixedType && !$type->isExplicitMixed()))
-				&& $type->isSuperTypeOf(TemplateTypeHelper::resolveToBounds($phpDocType))->yes()
+				($type->isCallable()->yes() && $phpDocType->isCallable()->yes())
```

Comment:
Alternatively, I could broaden the type of the callbacks in function map to `callable` and then I wouldn't need this. I'm not sure what's better.